cis_enforcements/gcp/cis_v200/section3/network.tf

```diff
@@ -0,0 +1,256 @@
+
+    APPROVE *
+  EOT
+}
+
+# GCP > Network > SSL Policy > Minimum TLS Version
+resource "turbot_policy_setting" "gcp_network_ssl_policy_minimum_tls_version" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/sslPolicyMinimumTlsVersion"
+  note     = "GCP CIS v2.0.0 - Control: 3.9"
+  value    = "Enforce: TLS 1.2"
+}
+
+# GCP > Network > SSL Policy > Profile
+resource "turbot_policy_setting" "gcp_network_ssl_policy_profile" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/sslPolicyProfile"
+  note     = "GCP CIS v2.0.0 - Control: 3.9"
+  value    = "Enforce: Restricted"
+}
+
+# GCP > Network > Firewall > Approved
+resource "turbot_policy_setting" "gcp_network_firewall_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/firewallApproved"
+  note     = "GCP CIS v2.0.0 - Control: 3.10"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved if new"
+}
+
+# GCP > Network > Firewall > Approved > Custom
+resource "turbot_policy_setting" "gcp_network_firewall_approved_custom" {
+  resource       = turbot_smart_folder.main.id
+  type           = "tmod:@turbot/gcp-network#/policy/types/firewallApprovedCustom"
+  note           = "Azure CIS v2.0.0 - Control: 3.10"
+  template_input = <<-EOT
+    {
+      firewall {
+        sourceRanges: get(path: "sourceRanges")
+        allowed: get(path: "allowed")
+      }
+    }
+  EOT
+  template       = <<-EOT
+    {%- set results = [] -%}
+
+    {%- if $.firewall.sourceRanges == null -%}
+
+      {%- set data = {
+          "title": "Source Ranges",
+          "result": "Not approved",
+          "message": "No source ranges defined"
+      } -%}
+
+    {%- elif $.firewall.sourceRanges != null -%}
+
+      {%- set allowedSourceIPRanges = ["35.235.240.0/20", "130.211.0.0/22", "35.191.0.0/16"] -%}
+      {%- set allRequiredIpExist = true -%}
+
+      {%- for ip in allowedSourceIPRanges -%}
+        {%- if ip not in $.firewall.sourceRanges -%}
+          {%- set allRequiredIpExist = false -%}
+        {%- endif -%}
+      {%- endfor -%}
+
+      {%- if allRequiredIpExist -%}
+
+        {%- set data = {
+            "title": "Source Ranges",
+            "result": "Approved",
+            "message": "All required source ranges are defined"
+        } -%}
+
+      {%- else -%}
+
+        {%- set data = {
+            "title": "Source Ranges",
+            "result": "Not approved",
+            "message": "Missing required source ranges"
+        } -%}
+
+      {%- endif -%}
+
+      {% set results = results.concat(data) -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "Source Ranges",
+          "result": "Skip",
+          "message": "No data for source ranges yet"
+      } -%}
+
+      {% set results = results.concat(data) -%}
+
+    {%- endif -%}
+
+    {%- if $.firewall.allowed == null -%}
```

Comment:
I think this is risky because it could set the approvedCustom to not approved if the CMDB control is not run yet. Can remove this check I think?